test/HttpApiClientTest.ts

```diff
@@ -0,0 +1,116 @@
+import {HttpApiClient, HttpClient} from "../src/httpClient";
+
+describe("HttpApiClient", () => {
+    let httpApiClient: HttpApiClient;
+
+    it("should return a valid baseUrl after construction", () => {
+
+        let defaultBaseUrl: string;
+
+        let httpClient = new HttpClient();
+        httpClient.configure((cfg) => {
+            defaultBaseUrl = cfg.options.baseUrl;
+        });
+
+        httpApiClient = new HttpApiClient();
+        httpApiClient.configure((cfg) => {
+            expect(cfg.options.baseUrl).toEqual(defaultBaseUrl);
+        });
+
+        httpApiClient = new HttpApiClient();
+        expect(httpApiClient.baseUrl).toBeUndefined();
+
+        httpApiClient = new HttpApiClient("https://myapi.crazy-factory.com");
+        expect(httpApiClient.baseUrl).toEqual("https://myapi.crazy-factory.com");
+
+    });
+
+    it("should throw errors for incorrect parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("asdf", "/users/1", undefined, undefined, undefined);
+        }).toThrowError();
+
+    });
+
+    it("should throw an error on incorrect GET parameters", () => {
+
+        expect(() => {
+            httpApiClient.request("GET", "/users/1", {id: 1}, undefined, undefined);
+        }).toThrowError();
+
+        expect(() => {
```

Comment:
Can be a separate tests a la "should not allow files for GET requests". You can use the .get() function directly here.

Optional: For good measure (and practice) you may also wrap all tests for files acceptance into a nested describe block. Same for data. Promise returns etc.

Alternatively: You can group by method and have the separate tests within the method group. 

(Grouping is done by having a describe() within a describe())